File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> pMap = new HashMap<>();
        
        for(String name : participant){
            pMap.put(name, pMap.getOrDefault(name, 0) + 1);
        }
        
        for(String comp : completion){
            pMap.put(comp, pMap.get(comp) - 1);
        }
        
        for(Map.Entry<String, Integer> entry: pMap.entrySet()){
            if(entry.getValue() == 0){
                continue;
            }
            return entry.getKey();
        }
        return null;
        
    
}
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    partimap = {}
    for part in participant:
        partimap[part] = partimap.get(part, 0) + 1
    for comp in completion:
        partimap[comp] = partimap.get(comp) - 1
    
    answer = max(partimap, key=partimap.get)
    return answer